From 9976cc4a356a759a1625e490161f107dcc560ade Mon Sep 17 00:00:00 2001
From: Christian Gagneraud <chgans@gna.org>
Date: Sat, 21 Apr 2012 10:34:17 +0100
Subject: [PATCH] Add missing attitude field initialisation in NMEA TNTHTM

Depth and Temperature are not initialised, causing the JSON output to
report a zero value instead of not reporting them at all

Signed-off-by: Eric S. Raymond <esr@thyrsus.com>
---
 driver_nmea0183.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/driver_nmea.c
+++ b/driver_nmea.c
@@ -915,6 +915,8 @@ static gps_mask_t processTNTHTM(int c UN
     session->gpsdata.attitude.acc_z = NAN;
     session->gpsdata.attitude.gyro_x = NAN;
     session->gpsdata.attitude.gyro_y = NAN;
+    session->gpsdata.attitude.temp = NAN;
+    session->gpsdata.attitude.depth = NAN;
     mask |= (ATTITUDE_SET);
 
     gpsd_report(LOG_RAW, "time %.3f, heading %lf (%c).\n",
